vars/mavenExecuteStaticCodeChecks.groovy

```diff
@@ -1,11 +1,45 @@
+import com.sap.piper.ConfigurationLoader
 import com.sap.piper.DownloadCacheUtils
+import com.sap.piper.QualityCheck
+import com.sap.piper.ReportAggregator
 import groovy.transform.Field
 
+import static com.sap.piper.Prerequisites.checkScript
+
 @Field String METADATA_FILE = 'metadata/mavenStaticCodeChecks.yaml'
 @Field String STEP_NAME = getClass().getName()
 
 void call(Map parameters = [:]) {
+    final script = checkScript(this, parameters) ?: null
     List credentials = []
-    parameters = DownloadCacheUtils.injectDownloadCacheInMavenParameters(parameters.script, parameters)
-    piperExecuteBin(parameters, STEP_NAME, METADATA_FILE, credentials)
+    parameters = DownloadCacheUtils.injectDownloadCacheInMavenParameters(script, parameters)
+
+    try {
+        piperExecuteBin(parameters, STEP_NAME, METADATA_FILE, credentials)
+    } catch (Exception exception) {
+        error("Maven Static Code Checks execution failed. Please examine the reports which are also available in the Jenkins user interface.")
+    }
+    finally {
+        showIssues(script)
+    }
+}
+
+private showIssues(Script script) {
+    Map configuration = ConfigurationLoader.stepConfiguration(script, STEP_NAME)
+    // Every check is executed by default. Only if configured with `false` the check won't be executed
+    if (!(configuration.spotBugs == false)) {
```

Comment:
Apparently this works, since it is covered by the Unit Test. So the expressions `!configuration.spotBugs` and `configuration.spotBugs == false` are indeed different for when the key `configuration.spotBugs` is not present?

Reply:
Mainly I want to avoid that null is resolved to false. That is why I compare the value with false and negate it afterwards

Reply:
I checked in a small test, the code is correct. It's just totally strange that via Groovy Truth the expression by itself "configuration.spotBugs" evaluates as a boolean (false) within an if-condition, yet comparing with a boolean "false", it works differently.